internal/provider/allocation.go

```diff
@@ -4,34 +4,34 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
+	"log"
 	"net/http"
 	"strings"
 
 	"terraform-provider-doit/internal/provider/models"
 	"terraform-provider-doit/internal/provider/resource_allocation"
-	"terraform-provider-doit/internal/provider/resource_allocation_group"
 
 	"github.com/hashicorp/terraform-plugin-framework/attr"
 	"github.com/hashicorp/terraform-plugin-framework/diag"
 	"github.com/hashicorp/terraform-plugin-framework/types"
 	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
 )
 
-func (plan *allocationResourceModel) toRequest(ctx context.Context) (models.SingleAllocation, diag.Diagnostics) {
+func (plan *allocationResourceModel) toRequest(ctx context.Context) (models.Allocation, diag.Diagnostics) {
 	var (
-		req   models.SingleAllocation
+		req   models.Allocation
 		diags diag.Diagnostics
 	)
 
-	allocationType := models.SingleAllocationAllocationType(plan.AllocationType.ValueString())
+	allocationType := models.AllocationAllocationType(plan.AllocationType.ValueString())
 	req.AllocationType = &allocationType
```

Comment:
totally cosmetic suggestion, consider defining req and assigning values to it within the same expression (potentially reducing variables sprawl in some cases), e.g. 
```
req := models.Allocation{ 
  AllocationType = &allocationType
  ...
}
```